crates/store/re_data_loader/src/mcap/schema/sensor_msgs/camera_info.rs

```diff
@@ -0,0 +1,214 @@
+use re_chunk::{
+    Chunk, ChunkId,
+    external::arrow::array::{FixedSizeListBuilder, Float64Builder, StringBuilder, UInt32Builder},
+};
+use re_log_types::TimeCell;
+use re_mcap_ros2::sensor_msgs;
+use re_sorbet::SorbetSchema;
+use re_types::{ComponentDescriptor, archetypes::Pinhole};
+
+use crate::mcap::{
+    cdr,
+    decode::{McapMessageParser, ParserContext, PluginError, SchemaName, SchemaPlugin},
+    schema::fixed_size_list_builder,
+};
+
+/// Plugin that parses `sensor_msgs/msg/CameraInfo` messages.
+#[derive(Default)]
+pub struct CameraInfoSchemaPlugin;
+
+impl SchemaPlugin for CameraInfoSchemaPlugin {
+    fn name(&self) -> SchemaName {
+        "sensor_msgs/msg/CameraInfo".into()
+    }
+
+    fn parse_schema(&self, _channel: &mcap::Channel<'_>) -> Result<SorbetSchema, PluginError> {
+        Err(PluginError::Other(anyhow::anyhow!("todo")))
+    }
+
+    fn create_message_parser(
+        &self,
+        _channel: &mcap::Channel<'_>,
+        num_rows: usize,
+    ) -> Box<dyn McapMessageParser> {
+        Box::new(CameraInfoMessageParser::new(num_rows)) as Box<dyn McapMessageParser>
+    }
+}
+
+pub struct CameraInfoMessageParser {
+    distortion_models: FixedSizeListBuilder<StringBuilder>,
+    k_matrices: FixedSizeListBuilder<Float64Builder>,
+    d_coefficients: Vec<Vec<f64>>,
+    widths: FixedSizeListBuilder<UInt32Builder>,
+    heights: FixedSizeListBuilder<UInt32Builder>,
+    binning_x: FixedSizeListBuilder<UInt32Builder>,
+    binning_y: FixedSizeListBuilder<UInt32Builder>,
+    frame_ids: FixedSizeListBuilder<StringBuilder>,
+    image_from_cameras: Vec<[f32; 9]>,
+    resolutions: Vec<(f32, f32)>,
+}
+
+impl CameraInfoMessageParser {
+    const ARCHETYPE_NAME: &str = "sensor_msgs.msg.CameraInfo";
+
+    pub fn new(num_rows: usize) -> Self {
+        Self {
+            distortion_models: fixed_size_list_builder(1, num_rows),
+            k_matrices: fixed_size_list_builder(9, num_rows),
+            d_coefficients: Vec::with_capacity(num_rows),
+            widths: fixed_size_list_builder(1, num_rows),
+            heights: fixed_size_list_builder(1, num_rows),
+            binning_x: fixed_size_list_builder(1, num_rows),
+            binning_y: fixed_size_list_builder(1, num_rows),
+            frame_ids: fixed_size_list_builder(1, num_rows),
+            image_from_cameras: Vec::with_capacity(num_rows),
+            resolutions: Vec::with_capacity(num_rows),
+        }
+    }
+}
+
+impl McapMessageParser for CameraInfoMessageParser {
+    fn append(&mut self, ctx: &mut ParserContext, msg: &mcap::Message<'_>) -> anyhow::Result<()> {
+        let sensor_msgs::CameraInfo {
+            header,
+            width,
+            height,
+            distortion_model,
+            d,
+            k,
+            r: _,
+            p: _,
+            binning_x,
+            binning_y,
+            roi: _,
```

Comment:
Instead of ignoring some fields (=throwing away user data), let's log them as custom components (like we already do for `distortion_model` etc).

We can also add a `TODO` to the relevant issue (maybe https://github.com/rerun-io/rerun/issues/2499 in this case?).